constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/Windows.java

```diff
@@ -137,6 +138,44 @@ public Windows shiftBy(Duration fromStart, Duration fromEnd){
     return ret;
   }
 
+  public Windows split(final int numberOfSubWindows) {
+    // Do nothing if N = 1
+    if (numberOfSubWindows == 1) {
+      return new Windows(this);
+    }
+
+    Windows ret = new Windows();
+    StreamSupport.stream(windows.ascendingOrder().spliterator(), false)
+      .forEach(x -> {
+        // Width of each sub-window, rounded down to the microsecond
+        final var width = Duration.divide(Duration.subtract(x.end, x.start), numberOfSubWindows);
+
+        // If there are fewer microseconds in the window than we can split, make one window per microsecond.
+        // This is likely only to come up for instantaneous windows, which we should leave unchanged.
+        final var numberOfMicroSeconds = Duration.subtract(x.end, x.start).in(Duration.MICROSECOND);
+        final int adjustedSubWindows;
+
+        if (x.isSingleton()) {
+          ret.add(Window.at(x.start));
+          return;
+        } else if (numberOfMicroSeconds < numberOfSubWindows) {
+          adjustedSubWindows = (int) numberOfMicroSeconds;
```

Comment:
Are we sure this behavior is something we want to do without informing the users? I can imagine an activity that does some value incrementation that they would want to operate exactly as many times as specified or be notified otherwise.

Reply:
@dyst5422 So should it throw an error if the window is too short? My general idea is that if the the window is too short, it is essentially like generating exclusive windows with 0 duration, which will get removed. I can throw an error instead if you think that makes more sense. But do you agree that splitting an instant should just return the instant unchanged?

Reply:
I have a feeling that the cases where this would be an issue (there aren't enough microseconds to split) are very small and it probably indicates some breakdown elsewhere if we're trying to split 3 microseconds into 4 activities. As such, I'm inclined to say we throw an error and double check with the users.

I haven't thought about the use cases as in depth as you have, but I struggle to identify a case where we'd be splitting an instantaneous activity that isn't similarly a breakdown in process elsewhere. The case of "I have an instantaneous 'event' that I want mapped into 4 different parallel activities" I think should be accomplished by other means/operators.

I prefer defaulting to the more conservative option (erroring in this case) as the safer solution as a general rule of thumb when there is uncertainty in the desired behavior (we apply this generously in SSG on APSS where we create errors anytime something isn't definite for example).

Reply:
I'm ok with this. I agree, having tiny windows like that is going to be at least an extremely rare, if not non-existent use case. For instants, I also bet it will be rare that users accidentally end up with a instant window mixed in with non-instant windows, so I guess its not a big deal. (It is possible though, if they use something like `resource.greaterThanOrEqual(2)` and the resource has a local maximum at exactly 2.)

I'll turn it into an error. If it becomes an issue for a user, we can reevaluate later.

Reply:
done